tests/frontend/test_program.py

```diff
@@ -630,3 +630,19 @@ def test_GBS_success(self):
         last_cmd = prog.circuit[-1]
         assert last_cmd.op.__class__ == ops.MeasureFock
         assert [x.ind for x in last_cmd.reg] == list(range(3))
+
+    def test_GBS_measure_fock_register_order(self):
+        """Test that compilation of MeasureFock on multiple modes
+        sorts the resulting register indices in ascending order."""
+        prog = sf.Program(4)
+
+        with prog.context as q:
+            ops.S2gate(0.45) | (q[0], q[2])
+            ops.S2gate(0.45) | (q[0], q[2])
+            ops.BSgate(0.54, -0.12) | (q[0], q[1])
+            ops.MeasureFock() | (q[0], q[3], q[2], q[1])
+
+        prog = prog.compile("gbs")
+        last_cmd = prog.circuit[-1]
+        assert last_cmd.op.__class__ == ops.MeasureFock
```

Comment:
how come not `isinstance`? seems to read better with an `assert`

Reply:
Good point :thinking: This was just a copy paste from the existing tests. I can change this.